sudo/src/pam.rs

```diff
@@ -1,15 +1,111 @@
-use sudo_common::error::Error;
+use std::fs::File;
 
-pub fn authenticate(username: &str) -> Result<(), Error> {
+use sudo_common::{error::Error, Context};
+use sudo_system::{
+    time::Duration,
+    timestamp::{RecordLimit, SessionRecordFile, TouchResult},
+    Process,
+};
+
+/// Tries to determine a record match limit for the current context.
+/// This should never produce an error since any actual error should just be
+/// ignored and no session record file should be used in that case.
+fn determine_record_limit(context: &Context) -> Option<RecordLimit> {
```

Comment:
The name `_limit` implies some time-based quantity where this is an "attribute-based" quantity; is there a better name possible? (No clear suggestion, scope?)